### ISSUE TITLE
[FEATURE] Improve Docs Page Navigation for Better User Experience

### ISSUE THREAD

**Issue by Shiva-Sai-ssb** [EddieHubCommunity/BioDrop#9517]
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

Currently, on the documentation page, the navigation on the left is not user-friendly. When a user picks a subcategory under a main category (such as "labels" in "Contributions"), the main category closes on its own, and the user has to open it again to choose a different subcategory.

Link : https://www.biodrop.io/docs/contributing/labels

### Proposed Solution

I'll give you an example to make it clear. I'm sharing a link to the Linkshub project. In this project, you'll see various categories on the left. If you select any sub-category within it, the main category will stay open until you choose to close it.

Link : https://www.linkshub.dev/

### Screenshots

Docs Page : 

![Screenshot (56)](https://github.com/EddieHubCommunity/BioDrop/assets/112751524/e1bac2a5-5ce1-4448-b08e-427dabb46362)

### Do you want to work on this issue?

Yes

### If you want to work on this issue...

I have fixed this problem locally, and now it works well on both desktop and mobile screens. To do this, I made changes to the sideNav.js file, including the one shown below:

```
<Disclosure as="div">
```

Replaced with :
```
<Disclosure defaultOpen={item.children.some(subItem => subItem.href === pathname)}>
```

I also made some other changes to it.

**Comment by amandamartin-dev** [comment]
Potentially related issue #9513

**Comment by SaraJaoude** [comment]
> Potentially related issue #9513

As #9513 may potentially resolve this, I will keep this issue in triage for the moment.

**Comment by JatinSainiOO7** [comment]
Hello!
Can I Work On This Issue.
THANKYOU.

**Comment by Abhishek-90** [comment]
@JatinSainiOO7 the issue is blocked for now. Until the issues is unblocked, the issue won't be assigned to anyone.

@SaraJaoude @amandamartin-dev shall we close chat for, until we decide what to do with this issue?

**Comment by SaraJaoude** [comment]
@Abhishek-90 agreed, thank you.  I will lock the conversation on this issue for now.

**Comment by SaraJaoude** [comment]
As #9513 has now been merged but this issue still persists, I am changing the label to `status: ready for dev`.

**Comment by ArvindParekh** [comment]
Hi, I'd be happy to give this a try if no one's working on it at the moment!

**Comment by SaraJaoude** [comment]
@Shiva-Sai-ssb as you raised this issue and requested to work on it, now that it is ready for dev - do you want this?  If not I will assign this to @ArvindParekh

**Comment by Shiva-Sai-ssb** [comment]
> @Shiva-Sai-ssb as you raised this issue and requested to work on it, now that it is ready for dev - do you want this? If not I will assign this to @ArvindParekh

Certainly! If @ArvindParekh is ready to contribute, please assign this issue to him.

**Comment by SaraJaoude** [comment]
Thanks @Shiva-Sai-ssb. @ArvindParekh  I will assign this to you.

**Comment by ArvindParekh** [comment]
Thanks @Shiva-Sai-ssb! Appreciate it. 
I was thinking of manually controlling the state of the disclosure, but I'll try your solution as well and see what works.

**Comment by eddiejaoude** [comment]
Great collaboration everyone 💪

**Comment by ArvindParekh** [comment]
Hey, I noticed somthing related to this issue.
I was thinking, could we make the navigation menu in the docs sticky? Something similar to this:


https://github.com/EddieHubCommunity/BioDrop/assets/71211731/23118137-0910-4c2a-af6d-ae5eccc251d8


It could make accessing links much easier when reading the docs, instead of having to scroll all the way up to find them. 
What do you think?
